astropy/convolution/convolve.py

```diff
@@ -20,24 +20,23 @@
 from ..modeling.core import _CompoundModelMeta
 from .utils import KernelSizeError, has_even_axis, raise_even_kernel_exception
 
-# Turn the faulthandler ON to help catch any signals, e.g. segfaults
-# or asserts. This doesn't, currently, work with Jupyter Notebook.
-faulthandler.enable()
-
 # Find and load C convolution library
-lib_path = glob.glob(os.path.join(os.path.dirname(__file__), 'lib_convolve*'))[0]
-if sys.platform.startswith('win'):
-    libConvolve = ctypes.windll.LoadLibrary(lib_path)
+lib_paths = glob.glob(os.path.join(os.path.dirname(__file__), 'lib_convolve*'))
+if len(lib_paths) > 0:
+    if sys.platform.startswith('win'):
```

Comment:
Think we need something like:
```python
...
#Search for library
lib_paths = glob.glob(os.path.join(root, 'lib_convolve*'))
lib_path =''
for path in lib_paths:
    if not path.endswith('.py'):
        lib_path = path
if not lib_path:
    raise Exception("Compiled convolution code is missing, try re-building astropy")

#Load found library
if sys.platform.startswith('win'):
    libConvolve = ctypes.windll.LoadLibrary(lib_path)
else:
    libConvolve = ctypes.cdll.LoadLibrary(lib_path)
...
```

Reply:
I think the exclusion soln. might be better than trying to be explicit for different OS lib suffixes. Any thoughts anyone?

Reply:
We can raise on ``if len(filtered_lib_paths) > 1``?

Reply:
What about .pyc files? We should make sure we include any kind of similar files. Any other extensions to exclude?

Reply:
I guess, though if the list gets too long then perhaps it's the wrong approach?

Reply:
Perhaps just ``*.py*``.

Reply:
pyo

Reply:
Actually, I'm not sure if `.pyd` would kill the windows usage...?

Reply:
Yep, we can't include ``.pyd``  in the exclusion list as this is what the the windows build produces.